_hardcoded/tracing.go

```diff
@@ -121,19 +123,19 @@ func MuxServerMiddleware(serviceName string) func(http.Handler) http.Handler {
 			}
 			ctx := r.Context()
 
-			var crid string
-
 			s := trace.SpanFromContext(ctx)
-			crid = s.SpanContext().TraceID().String()
 			bags := baggage.FromContext(ctx)
-			// Add clever-request-id to baggage
-			member, err := baggage.NewMember("clever-request-id", crid)
-			if err != nil {
-				logger.FromContext(ctx).WarnD("error-creating-baggage", logger.M{"error": err.Error()})
-			} else {
-				bags, err = bags.SetMember(member)
+
+			if bags.Member("clever-request-id").String() == "=" { // if clever-request-id is not set
```

Comment:
Otel is dumb and they don't have a way of checking if a baggage key exists. Their get returns an "empty member" if it doesn't exist which isn't nil or "" so you can't compare to something simple like that. If you attempt to get the value of something that doesn't exist though you get a nil pointer dereference. Which makes no sense to me. 

So to check if something doesn't exist the safe/low effort way of doing it is call their String() function on it which returns `<key>=<value>` and if this member doesn't exist that ends up just being `=`

You can shoot me now.